src/abstractdataframe/nest.jl

```diff
@@ -0,0 +1,390 @@
+   1 │ b       3×2 DataFrame 
+   2 │ a       2×2 DataFrame 
+   3 │ c       1×2 DataFrame 
+
+julia> n2.zx
+3-element Vector{DataFrame}:
+ 3×2 DataFrame
+ Row │ z     x     
+     │ Char  Int64 
+─────┼─────────────
+   1 │ a         1
+   2 │ e         5
+   3 │ f         6
+ 2×2 DataFrame
+ Row │ z     x     
+     │ Char  Int64 
+─────┼─────────────
+   1 │ b         2
+   2 │ c         3
+ 1×2 DataFrame
+ Row │ z     x     
+     │ Char  Int64 
+─────┼─────────────
+   1 │ d         4
+
+julia> n3 = nest(groupby(df, :id), :x => "x", [:y, :z] => "yz")
+3×3 DataFrame
+ Row │ id      x              yz
+     │ String  DataFrame      DataFrame     
+─────┼──────────────────────────────────────
+   1 │ b       3×1 DataFrame  3×2 DataFrame 
+   2 │ a       2×1 DataFrame  2×2 DataFrame 
+   3 │ c       1×1 DataFrame  1×2 DataFrame 
+
+julia> n3.x
+3-element Vector{DataFrame}:
+ 3×1 DataFrame
+ Row │ x     
+     │ Int64 
+─────┼───────
+   1 │     1
+   2 │     5
+   3 │     6
+ 2×1 DataFrame
+ Row │ x     
+     │ Int64 
+─────┼───────
+   1 │     2
+   2 │     3
+ 1×1 DataFrame
+ Row │ x     
+     │ Int64 
+─────┼───────
+   1 │     4
+
+julia> n3.yz
+3-element Vector{DataFrame}:
+ 3×2 DataFrame
+ Row │ y      z    
+     │ Int64  Char 
+─────┼─────────────
+   1 │    11  a
+   2 │    15  e
+   3 │    16  f
+ 2×2 DataFrame
+ Row │ y      z    
+     │ Int64  Char 
+─────┼─────────────
+   1 │    12  b
+   2 │    13  c
+ 1×2 DataFrame
+ Row │ y      z    
+     │ Int64  Char
+─────┼─────────────
+   1 │    14  d
+"""
+nest(gdf::GroupedDataFrame, cols::Pair{<:Any, <:AbstractString}...) =
+    combine(gdf, (sdf -> (; Symbol(dst) => select(sdf, index(sdf)[src]))
+                  for (src, dst) in cols)...)
+nest(gdf::GroupedDataFrame, cols::Pair{<:Any, Symbol}...) =
+    combine(gdf, (sdf -> (; dst => select(sdf, index(sdf)[src]))
+                  for (src, dst) in cols)...)
+nest(gdf::GroupedDataFrame) = nest(gdf, valuecols(gdf) => :data)
+
+"""
+    unnest(df::AbstractDataFrame, src::ColumnIndex...;
+           cols::Union{Symbol, AbstractVector{Symbol},
+                       AbstractVector{<:AbstractString}}=:setequal,
+           promote::Bool=true,
+           makeunique::Bool=false, flatten::Bool=true)
+
+Unnest one or more columns `src` into multiple columns. The newly created
+columns are stored at the end of the data frame (and the `src` column is
+dropped).
+
+Each `src` column must contain a `NamedTuple`, a `DataFrameRow`, a
+`Tables.AbstractRow`, or Tables.jl table.
+
+`cols` (default `:setequal`) and `promote` (default `true`) keyword arguments
+have the same meaning as in [`push!`](@ref).
+
+If `makeunique=false` (the default) produced column names must be unique.
+If `makeunique=true` then duplicate column names will be suffixed with `_i`
+(`i` starting at `1` for the first duplicate).
+
+If `flatten=true` (the default) then newly created columns are flattened
+using [`flatten`](@ref) with `scalar=Missing` keyword argument.
```

Comment:
Why do we need to special case `Missing`?

Reply:
we need because if in some table column is missing we get `missing` initially, which has to get expanded. Now this text is removed.